contracts/core/src/libraries/Secp256k1.sol

```diff
@@ -45,4 +76,75 @@ library Secp256k1 {
         // Verify the derived point lies on the curve
         return EllipticCurve.isOnCurve(x, y, AA, BB, PP);
     }
+
+    /**
+     * @notice Convert public key coordinates to an Ethereum address
+     * @param x The x coordinate of the public key
+     * @param y The y coordinate of the public key
+     * @return The Ethereum address corresponding to the public key
+     */
+    function publicKeyToAddress(bytes32 x, bytes32 y) internal pure returns (address) {
+        bytes memory pubKey = new bytes(64);
+        assembly {
+            // Store x and y coordinates
+            mstore(add(pubKey, 32), x)
+            mstore(add(pubKey, 64), y)
+        }
+
+        // Hash the public key and keep last 20 bytes
+        bytes32 hash = keccak256(pubKey);
+        return address(uint160(uint256(hash)));
+    }
+
+    /**
+     * @notice Get the Ethereum signed message hash of a public key
+     * @param x The x coordinate of the public key
+     * @param y The y coordinate of the public key
+     * @return The Ethereum signed message hash
+     */
+    function getEthSignedMessageHash(bytes32 x, bytes32 y) internal pure returns (bytes32) {
+        // Use full uncompressed public key as the message
+        bytes memory message = abi.encodePacked(hex"04", x, y);
+        bytes32 ethSignedMessageHash =
+            keccak256(abi.encodePacked("\x19Ethereum Signed Message:\n32", keccak256(message)));
+        return ethSignedMessageHash;
+    }
+
+    /**
+     * @notice Verify a signature against a public key
+     * @param x The x coordinate of the public key
+     * @param y The y coordinate of the public key
+     * @param signature The 65-byte signature (r,s,v)
+     * @return True if signature is valid, false otherwise
+     */
+    function verifySignature(bytes32 x, bytes32 y, bytes calldata signature) internal pure returns (bool) {
+        require(signature.length == 65, "Secp256k1: invalid signature length");
+        bytes32 ethSignedMessageHash = getEthSignedMessageHash(x, y);
+
+        // Extract signature components
+        bytes32 r;
+        bytes32 s;
+        uint8 v;
+        assembly {
+            r := calldataload(signature.offset)
+            s := calldataload(add(signature.offset, 32))
+            v := byte(0, calldataload(add(signature.offset, 64)))
+        }
+
+        // v needs to be 0 or 1 for secp256k1 (not 27/28 like Ethereum)
+        require(v == 0 || v == 1, "Secp256k1: invalid v value");
+
+        // Add 27 to make it compatible with ecrecover
+        v += 27;
```

Comment:
rather than translating here, i suggest we require one (probably 27 / 28) impose that requirement on signers (our lib/k1util::Sign already produces signatures with v 27/28)

that said I think we should just use a oz's escda library for this. It does all these things and others (like keeping s value to lower bound, to prevent malleable signatures).

best to default to battle tested / audited code when we can, particularly for crypography related stuff